pkg/common/common.go

```diff
@@ -103,3 +109,14 @@ func ValidateNodePoolLabels(nodePoolName string, labels map[string]string) error
 
 	return nil
 }
+
+// ValidateNodePoolName checks the validity of the node pool name
+func ValidateNodePoolName(nodePoolName string) error {
+	if matched, err := regexp.MatchString(nodePoolNameValidationRegexp, nodePoolName); !matched {
+		return errors.New(fmt.Sprintf("node pool name '%s' invalid: a DNS-1123 subdomain must consist of lower case alphanumeric characters", nodePoolName))
+	} else if err != nil {
+		return errors.WrapIf(err, "couldn't validate node pool name")
+	}
```

Comment:
**Recommended**: it is general practice to check error first and matching later, because errors can screw with results. In this example when the regex returns an error the matched value is probably left untouched which means the it will be the default value which is false for boolean so the first if branch will always be executed even when there is an error.

Also for static regexes it might be worth saving them as a compiled regex with `regexp.MustCompile()` and use their member functions for matches, because that cannot generate any error so the logic gets easier. The must compilation can fail and it throws a panic on call (initialization time for globals) which is generally avoided, that's why I advise it only for static regexes (where any issue would be brought to light during initial testing).

Reply:
Thanks, will do!